### PR TITLE
Demo script squashed

### LINKED ISSUE #13: implement demo bash script as end2end test
demonstrate the following scenarios

Scenario 1: Happy flow
1.	Alice and Bob both create an account and receive funds from faucet (some native tokens as well as 200 tokenA for Alice and 200 tokenB for Bob)
2.	Alice and Bob both create and register a proxy account
3.	Alice deposits 100 tokenA 
4.	Bob deposits 100 tokenB
5.	Alice places a limit order selling 50 tokenA at a limit of 40 tokenB
6.	Bob places a limit order buying 50 tokenA at a limit of 60 tokenB
7.	The matching engine clears the match, sends it to the gateway
8.	The gateway settles the match, publishes all details
9.	The offchain balance of Alice is 50 tokenA plus 50 tokenB
10.	The offchain balance of Bob is 50 tokenA plus 50 tokenB
11.	Alice withdraws all her tokenB through direct call to gateway
12.	Bob withdraws all his tokenA through indirect extrinsic
13.	The offchain balance of Alice is zero tokenB and Bob is zero tokenA
14.	The onchain balance of Alice is 50 tokenB and Bob is 50 tokenA

Scenario 2: 
Along scenario 1 with the differences of:
•	Bob submits an order buying 50 tokenA at a limit of 110 tokenB
•	The order gets rejected by the gateway because it is not backed by Bobs balance

Scenario 3: 
Along scenario 1 with the differences of:
•	After the settled trade, Bob tries to withdraw 60 tokenB
•	The withdrawal gets rejected by the gateway because it is not backed by Bobs balance

Scenario 4:
Along scenario 1 with the differences of:
•	Matching engine returns a match involving an order that hasn’t been sent through the gateway
•	The match gets rejected by the gateway

### PR DESCRIPTION
**Pull request by haerdib**
- adds the happy flow demo script
- add scenarios (unhappy flows) 2 and 3
- updates make update with specific jsonrpc commit (otherwise cargo update breaks)
- add token balance client call

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -225,10 +225,13 @@ mrsigner:
 
 .PHONY: update
 update:
+
 	@echo "Running cargo update.."
 	@cargo update
 	@echo "cargo update in enclave directory"
 	@cd enclave && cargo update
+	@cd enclave && cargo update -p jsonrpc-core --precise e5ee60bc30dedf513743843be2523dc384bbcae1
```
Comment (Gauthamastro) [inline Makefile:233]: why is there a duplicate?

**Comment by Gauthamastro** [comment]
I need the weekend to go over this PR

File: Makefile (hunk 1 above)
Comment (haerdib) [reply Makefile:233]: To ensure it has updated & removed all obsolete crates correctly.

Comment (Gauthamastro) [reply Makefile:233]: ok

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**